05_testing_and_ci/examples/python-testing/test_operations.py

```diff
@@ -0,0 +1,52 @@
+"""
+Tests for mathematical operations functions.
+"""
+
+from operations import find_max, find_average
+import pytest
+import csv
+
+
+# Unit test
+def test_find_max():
+    data = [43, 32, 167, 18, 1, 209]
+    expected_result = 209
+    actual_result = find_max(data)
+    assert actual_result == expected_result
+
+
+# Unit test
+def test_find_average():
+    data = [43, 32, 167, 18, 1, 209]
+    expected_result = 78.33
+    # expected_result = pytest.approx(78.3, abs=0.01)
+    actual_result = find_average(data)
+    assert actual_result == expected_result
+
+
+# Integration test
+def test_max_avg_compare():
+    """
+    Test to check that the average of a data set is always lesser than the maximum
+    """
+    data = [43, 32, 167, 18, 1, 209]
+    maximum = find_max(data)
+    average = find_average(data)
+    assert average < maximum
```

Comment:
I am bit confused by what you are suggesting here. So the `reference_average` is given by the user then?

Reply:
Yes, the test average would be given by the user. My idea was that one "integrates" the functions more tightly with each other such that one of the functions uses the output of another function. From my understanding, this would be closer to classical integration testing, but maybe I understood something wrong about integration testing.